pyproject.toml

```
@@ -10,4 +10,102 @@ requires = [
 ]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "cheroot"
+description = "Highly-optimized, pure-python HTTP server"
+authors = [
+    { name = "CherryPy Team", email = "team@cherrypy.dev" },
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: Web Environment",
+    "Intended Audience :: Developers",
+    "Operating System :: OS Independent",
+    "Framework :: CherryPy",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: Implementation",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Programming Language :: Python :: Implementation :: Jython",
+    "Programming Language :: Python :: Implementation :: PyPy",
+    "Topic :: Internet :: WWW/HTTP",
+    "Topic :: Internet :: WWW/HTTP :: HTTP Servers",
+    "Topic :: Internet :: WWW/HTTP :: WSGI",
+    "Topic :: Internet :: WWW/HTTP :: WSGI :: Server",
+    "Typing :: Typed",
+]
+keywords = [
+    "http",
+    "server",
+    "ssl",
+    "wsgi",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "importlib_metadata; python_version < '3.8'",
+    "more_itertools >= 2.6",
+    "jaraco.functools",
+]
+dynamic = ["version"]
+
+[project.urls]
+Homepage = "https://cheroot.cherrypy.dev"
+"Chat: Matrix" = "https://matrix.to/#/#cherrypy-space:matrix.org"
+"CI: GitHub" = "https://github.com/cherrypy/cheroot/actions"
+"Docs: RTD" = "https://cheroot.cherrypy.dev"
+"GitHub: issues" = "https://github.com/cherrypy/cheroot/issues"
+"GitHub: repo" = "https://github.com/cherrypy/cheroot"
+"Tidelift: funding" = "https://tidelift.com/subscription/pkg/pypi-cheroot?utm_source=pypi-cheroot&utm_medium=referral&utm_campaign=pypi"
+
+[project.readme]
+file = "README.rst"
+content-type = "text/x-rst"
+
+[project.optional-dependencies]
+docs = [
+    # upstream
+    "sphinx>=1.8.2",
+    "jaraco.packaging>=3.2",
+    "sphinx-tabs>=1.1.0",
```

This is a bit inconsistent with the preferred style used elsewhere so let's take this opportunity to make the specs sparse:
```suggestion
    "sphinx >= 1.8.2",
    "jaraco.packaging >= 3.2",
    "sphinx-tabs >= 1.1.0",
```